processor/src/decorators/mod.rs

```diff
@@ -83,6 +84,27 @@ where
         Ok(())
     }
 
+    /// Merges two Merkle trees with roots on the top two words of the operand stack.
+    ///
+    /// The operand stack is expected to be arranged as follows (from the top):
+    /// - root of the right tree, 4 elements
+    /// - root of the left tree, 4 elements
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - Any of the input roots (left or right) doesn't exist as a Merkle tree for the advice
+    /// provider.
+    fn inject_merkle_merge(&mut self) -> Result<(), ExecutionError> {
+        // fetch the arguments from the stack
+        let lhs = [self.stack.get(7), self.stack.get(6), self.stack.get(5), self.stack.get(4)];
+        let rhs = [self.stack.get(3), self.stack.get(2), self.stack.get(1), self.stack.get(0)];
+
+        // perform the merge
+        self.advice_provider.merge_roots(lhs, rhs)?;
+
+        Ok(())
```

Comment:
nit: should this be just:
```Rust
self.advice_provider.merge_roots(lhs, rhs)
```

Reply:
We can't do that one because `AdviceProvider::merge_roots` returns the new root `Word`, and the inject functions are expected to return unit `()`